mail_tracking/models/ir_mail_server.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import re
+from openerp import models, api, tools
+
+
+class IrMailServer(models.Model):
+    _inherit = "ir.mail_server"
+
+    def _tracking_headers_add(self, tracking_email_id, headers):
+        """Allow other addons to add its own tracking SMTP headers"""
+        headers = headers or {}
+        headers['X-Odoo-Tracking-ID'] = tracking_email_id
+        return headers
+
+    def _tracking_email_id_body_get(self, body):
+        body = body or ''
+        tracking_email_id = False
+        # https://regex101.com/r/lW4cB1/2
+        match = re.search(
+            r'<img [^>]* data-odoo-tracking-email=["\']([0-9]*)["\']', body)
```

Comment:
re.search is more efficient in this case and this img attribute is specific enougth, IMO, in order to guarantee  that there will be only one in a email body.